pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -217,28 +288,50 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 						logger.Infof("skipping event from node %s status update since it is already completed", node)
 						continue
 					}
-					completed := c.handleStatusConfigMapStatus(node, config, configMap, configOSDs)
+					var completed bool
+					completed, deferredConfigMaps = c.handleStatusConfigMapStatus(node, config, configMap, deferredConfigMaps, configOSDs)
 					if completed {
 						remainingNodes.Remove(node)
 						if remainingNodes.Count() == 0 {
 							logger.Infof("%d/%d node(s) completed osd provisioning", originalNodes, originalNodes)
+							handleDeferredConfigMaps()
 							return true
 						}
 					}
 				}
 
+			case <-opportunityTimer.C:
+				// We want to make a best-effort attempt to update configmaps while provisioning is
+				// ongoing. While there are deferred configmaps remaining, grab one and process it.
+				if len(deferredConfigMaps) > 0 {
+					configMap := deferredConfigMaps[0]
+					logger.Debugf("opportunistically handling deferred OSD status %q", configMap.Name)
+
+					node, ok := configMap.Labels[nodeLabelKey]
+					if !ok {
+						logger.Infof("missing node label on deferred configmap %s", configMap.Name)
+						continue
+					}
+
+					c.handleDeferredStatusConfigMapStatus(node, config, configMap, configOSDs)
+
+					// remove the configmap we just processed from the deferred configmap list
+					deferredConfigMaps = deferredConfigMaps[1:]
+				}
+
 			case <-time.After(time.Minute):
```

Comment:
I will double check this, but this case should still be reached. I think it would only be a problem if the `<-time.After(time.Minute)` were reset with every loop, which it isn't.

Reply:
Ah, I assumed it would be reset with every loop

Reply:
You're right. It *is* reset with every loop. I thought the previous behavior was to always log every minute, but I see it's only logging every minute that we spend waiting. I'll have to figure out a way to rework this so that it will only time when we are *not* making progress.

Reply:
I fixed it by keeping track of whether the loop is making progress with `shouldResetTimeoutTimer`.